### ISSUE TITLE
Invalid HTML structure in search result item

### ISSUE THREAD

**Issue by ggeisler** [projectblacklight/arclight#1395]
Within the search result item structure there is this code for displaying the breadcrumb links/path to the item:

```html
<dl class="document-metadata dl-invert row">
    <div class="breadcrumb-links">
         <ol class="breadcrumb">
           <li class="breadcrumb-item">
              ...
```

This raises an accessibility error, and is also just invalid HTML, I believe. The `dl` can only contain `dt` and `dd` elements, so the child `div` is invalid, as is the subsequent `ol`. 

Probably fixable just by changing the `dl` to a `div`? However, if the result item has metadata to display aside from the breadcrumb path (e.g., an abstract/scope value), that part is also within the `dl` and includes an additional invalid `div`:

```html
<dl class="document-metadata dl-invert row">
    <div class="row">
       <dt class="visually-hidden">Abstract Or Scope</dt>
         <dd class="col al-document-abstract-or-scope truncator truncated" data-controller="arclight-truncate">
            <div class="content" data-arclight-truncate-target="content"><p>"Broadcasting", mainly ...
           ...
```

Basically it seems we're using a `dl` for the result item metadata following the basic Blacklight pattern (where the `dl` is used correctly and appropriately for displaying several field labels and values), but ArcLight result items display very different item metadata, so the `dl` ends up interspersed with a lot of invalid elements.

Perhaps the whole result item structure in ArcLight should be rewritten (ignoring the Blacklight way of doing it) with an eye on valid HTML.

The code in question is what renders this:

<img width="852" alt="Screen Shot 2023-01-24 at 11 01 24 AM" src="https://user-images.githubusercontent.com/101482/214372261-fcabad1f-7a03-4af4-8f81-440afd9772e7.png">

**Comment by cbeer** [comment]
![Screen Shot 2023-01-30 at 18 30 40](https://user-images.githubusercontent.com/111218/215646882-78bc0a3f-eca5-4b13-9db4-8763e7a0e65a.png)

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl

🤔

**Comment by ggeisler** [comment]
Ah, interesting. One of the accessibility checking tools I used (Axe DevTools) says this:

> dl element has direct children that are not allowed: div > ol

Either way, based on the documentation, yes, I think the second part of my initial description that says we shouldn't have a `div` within the `dl` is incorrect.

I guess I'm not completely clear on whether the first part -- whether the `dl` can have a `div` that then contains an `ol` -- is still true, or whether the Axe DevTools is incorrect (or maybe is using a different/outdated spec for determining what is allowable).

Regardless, this doesn't seem to have a negative effect on the UI, and the other accessibility tools I used didn't mention this issue, so perhaps I should close this ticket?

**Comment by cbeer** [comment]
> A name-value group consists of one or more names ([dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) elements, possibly as children of a [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element child) followed by one or more values ([dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) elements, possibly as children of a [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element child), ignoring any nodes other than [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) and [dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) element children, and [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) and [dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) elements that are children of [div](https://html.spec.whatwg.org/multipage/grouping-content.html#the-div-element) element children.

---

> If the element is a child of a [dl](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) element: one or more [dt](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dt-element) elements followed by one or more [dd](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dd-element) elements, optionally intermixed with [script-supporting elements](https://html.spec.whatwg.org/multipage/dom.html#script-supporting-elements-2).

🤔 . It sounds like a `dl > div` isn't the same as a regular old div... so, I guess I agree we should do something better?

**Comment by dinahhandel** [comment]
Needs developer investigation but can be worked on.

**Comment by lfarrell** [comment]
Fix PR: https://github.com/projectblacklight/arclight/pull/1447


    * Also addresses color contrast issues and duplicate ids in search results
    * Also addresses color contrast issue in highlighted item in expandable hierarchy menu

Should fix all accessibility issues in arclight.
